### ISSUE TITLE
2020-04-13 tc158 M1SS AUC0_10 + AUC10_24 doesn't match AUCLAST where TLAST == 24

### ISSUE THREAD

**Issue by TomTensfeldt** [tensfeldt/openNCA#228]
Partial AUCs AUC0_10 and AUC10_24 do not match the AUCLAST which is expected to be identical to AUC0_24. 

mct$TIME == "NOMINAL"

testcase tc158 M1SS is on the onedrive folder for this.

```r
> mct$TIME <- "NOMINAL"
> results_list <- list()
> results_list <- run_computation(data=d, map=mct, flag=flags, parameterset=parameterset)
unit_conversion : Unit Class 1 (Time) time_col:  TMAX TMAX1 TMAX2 TMIN TMIN1 TMIN2 TLAST TLAST1 TLAST2 TLAG KELTMLO KELTMHI THALF LASTTIME LASTTIME1 LASTTIME2 MRTLAST MRTLAST1 MRTLAST2 MRTEVIFO1 MRTEVIFO2 MRTEVIFP1 MRTEVIFP2 TAU1 TAU2 TOLD1 TOLD2  parameters are scaled from  HR  to  HR  via scaling factor:  1 
unit_conversion : Unit Class 3 (Dose) dose_col:  DOSE1  parameters are scaled from  MCG  to  MG  via scaling factor:  0.001 
unit_conversion : Unit Class 3 (Dose) dose_col:  DOSE2  parameters are scaled from  MCG  to  MG  via scaling factor:  0.001 
unit_conversion : Unit Class 4 (Volume) volume_col:  VZFTAU1 VZFTAU2  parameters are scaled from  ML  to  L  via scaling factor:  1 
unit_conversion : Unit Class 5 (Amount/Volume) conc_col:  CMAX CMAX1 CMAX2 CMIN CMIN1 CMIN2 CLAST CLAST1 CLAST2 CTROUGH1 CTROUGH2 CTROUGHEND1 CTROUGHEND2 KELC0 CEST CAV1 CAV2  parameters are scaled from  NG/ML  to  NG/ML  via scaling factor:  1 
unit_conversion : Unit Class 6 (1/Time) kel_col:  KEL  parameters are scaled from  1/HR  to  1/HR  via scaling factor:  1 
unit_conversion : Unit Class 7 (Volume/Time) cl_col:  CLFTAU1 CLFTAU2  parameters are scaled from  ML/HR  to  L/HR  via scaling factor:  1 
unit_conversion : Unit Class 8: (Amount.Time/Volume) auc_col:  AUCALL AUCLAST AUCLASTC AUCLAST1 AUCLAST2 AUCLASTC1 AUCLASTC2 AUC1 AUC2 AUC3 AUC4 AUC5 AUC6 AUC7 AUC8 AUC9 AUC10 AUC0_10 AUC10_24 AUCINFO AUCINFO1 AUCINFO2 AUCINFOC AUCINFP AUCINFP1 AUCINFP2 AUCINFPC AUCTAU1 AUCTAU2  parameters are scaled from  NG.HR/ML  to  NG.HR/ML  via scaling factor:  1 
unit_conversion : Unit Class 9: (Amount.Time.Time/Volume) aumc_col:  AUMCLAST1 AUMCLAST2 AUMCINFO1 AUMCINFO2 AUMCINFP1 AUMCINFP2 AUMCTAU1 AUMCTAU2  parameters are scaled from  NG.HR.HR/ML  to  NG.HR.HR/ML  via scaling factor:  1 
unit_conversion : Unit Class 12: ([Amount/Volume]/Amount) concdn_col:  CMAXDN CMAXDN1 CMAXDN2 CMINDN CMINDN1 CMINDN2  parameters are scaled from  NG/ML/MCG  to  NG/ML/MG  via scaling factor:  1000 
unit_conversion : Unit Class 14: (Volume/Body Weight) volumew_col:  VZFTAUW1 VZFTAUW2  parameters are scaled from  ML/LB  to  L/KG  via scaling factor:  2.204624 
unit_conversion : Unit Class 15: (Volume/Time/Body Weight) clw_col:  CLFTAUW1 CLFTAUW2  parameters are scaled from  ML/HR/LB  to  ML/HR/KG  via scaling factor:  2.204624 
There were 12 warnings (use warnings() to see them)
> r <- results_list$data_out
> e <- results_list$est_data
> f <- results_list$flag_data
> chkresults(dataset=d, mct=mct, parameterset=parameterset, results=r)
parameterset:  PARAMETERLIST 
[1] "AUCDN"    "AUCT"     "AUCT1_T2" "AUCTDN"  
> 
> mct[,names(parameter_required("^AUC.[0-9]+?.T[1-2]$", names(mct), simplify=FALSE))]
  AUC.1.T1 AUC.1.T2 AUC.2.T1 AUC.2.T2
1        0       10       10       24
> 
> # Check Partial AUCs
> x.tst <- r$AUC0_10 + r$AUC10_24
> x.ref <- r$AUCLAST
> compare(x.tst, x.ref, tolerance=tolerance)
39/39 mismatches (average diff: 97.5)
[1] 306 - 153.2 == 153.2
[2] 210 - 105.2 == 105.2
[3] 149 -  74.6 ==  74.6
[4] 259 - 129.4 == 129.4
[5] 413 - 206.6 == 206.6
[6] 155 -  77.5 ==  77.5
[7]  99 -  49.5 ==  49.5
[8] 141 -  70.6 ==  70.6
[9] 290 - 144.8 == 144.8
...> 
> # Note that TLAST is 24 hours
> unique(r$TLAST)
[1] 24
```

**Comment by jhhughes256** [comment]
Interestingly the sum of the partial AUC from 0 to 10 and 10 to 24 is very close to the dose-normalised incremental AUC from 0 to 24.

```r
> paste(testcase.id, mct$MODEL, mct$DOSINGTYPE, mct$AUCMETHOD, mct$TIME, sep = ".")
[1] "tc158.M1.SS.LINLOG.ACTUAL"
> r$AUC0_10 + r$AUC10_24
 [1] 306.29469 210.51625 149.07787 264.77965 413.22422 154.92129  98.97724 141.25926 289.54648  96.80353 216.24768
[12] 294.41169 112.48857  99.74646 148.00567 120.76106 132.60619 137.26645 128.02057 166.50616 110.44490 249.59594
[23] 142.21279 339.96976 100.71537 128.49476 308.02008 144.25361 433.07302 293.88264 194.63723 210.36752 143.31988
[34] 137.53472 185.74462 158.89782 172.11937 150.41128 288.90078
> r$AUC10DN
 [1] 305.48615 210.14150 148.94816 264.46582 412.64735 154.82093  98.89581 141.10859 289.12703  96.73204 216.04746
[12] 294.06660 112.41109 135.72952 147.87667 120.65818 132.28146 137.11227 127.91887 166.37827 110.34451 248.81429
[23] 142.21261 339.50616 100.31704 128.37972 306.85268 144.14420 432.38399 293.44864 194.44291 210.15107 143.19451
[34] 137.42940 185.23098 158.76864 171.81787 150.04601 288.77597
> unique(r$AUCINT10)
[1] "0_23.9166666666667"                  "0_23.9666666666667"                  "0.0166666666666667_23.9666666666667"
[4] "0_24"                                "0_23.85"                             "0_23.8833333333333"   
```

**Comment by jhhughes256** [comment]
With commit b76b846 this issue cannot be tested as a result of #234

**Comment by jhhughes256** [comment]
This issue appears to have been resolved as of commit 9994c29. See output below.

```r
> paste(testcase.id, mct$MODEL, mct$DOSINGTYPE, mct$AUCMETHOD, mct$TIME, sep = ".")
[1] "tc158.M1.SS.LINLOG.NOMINAL"
> r$AUC0_10 + r$AUC10_24
 [1] 153.20597 105.17590  74.56051 129.35063 206.59636  77.49714  49.50462  70.62225 144.71161  48.42054 108.15904 147.17434  56.27141  68.02167  74.03175
[16]  60.40012  66.19613  68.64855  64.03397  83.27707  55.23380 124.84388  71.11556 169.96912  50.50327  64.26891 154.10505  72.15766 216.47551 146.87014
[31]  97.35341 105.21575  71.68049  68.79544  92.90219  79.47827  86.02384  75.23809 144.55133
> r$AUC10
 [1] 153.20597 105.17590  74.56051 129.35063 206.59636  77.49714  49.50462  70.62225 144.71161  48.42054 108.15904 147.17434  56.27141  68.02167  74.03175
[16]  60.40012  66.19613  68.64855  64.03397  83.27707  55.23380 124.84388  71.11556 169.96912  50.50327  64.26891 154.10505  72.15766 216.47551 146.87014
[31]  97.35341 105.21575  71.68049  68.79544  92.90219  79.47827  86.02384  75.23809 144.55133
> unique(r$AUCINT10)
[1] "0_24"
> r$AUCLAST
 [1]        NA 105.17590        NA 129.35063 206.59636        NA        NA        NA 144.71161        NA        NA 147.17434        NA        NA        NA
[16]        NA  66.19613        NA        NA        NA        NA 124.84388        NA 169.96912        NA        NA 154.10505        NA 216.47551 146.87014
[31]        NA        NA        NA        NA        NA        NA  86.02384        NA 144.55133
> r$AUCLAST_test <- r$AUCLAST
> r$AUCLAST_test[is.na(r$AUCLAST)] <- r$AUCLAST1[is.na(r$AUCLAST)]
> r$AUCLAST_test
 [1] 153.20597 105.17590  74.56051 129.35063 206.59636  77.49714  49.50462  70.62225 144.71161  48.42054 108.15904 147.17434  56.27141  68.02167  74.03175
[16]  60.40012  66.19613  68.64855  64.03397  83.27707  55.23380 124.84388  71.11556 169.96912  50.50327  64.26891 154.10505  72.15766 216.47551 146.87014
[31]  97.35341 105.21575  71.68049  68.79544  92.90219  79.47827  86.02384  75.23809 144.55133
```
